src/utils/utf8.c

```diff
@@ -152,29 +182,53 @@ utf8_toupper(utf8_ctx_t *ctx, const char *s, char *buf, size_t buflen)
 	return len;
 }
 
+
+ssize_t
+utf8_subs_diacritics(utf8_ctx_t *ctx, UChar *buf, ssize_t buflen)
```

Comment:
- The public API (of our little UTF-8 abstraction) must not expose the libicu internals. That is the purpose of wrapping this.
- You need to take UTF-8 string, convert it into UTF-16 and convert it back to UTF-8. The UTF-8 should be in `strbuf_t` (our string buffer abstraction).

Basically, it's the same kind of machinery like in the `utf8_normalize()` function.